Mathlib/RingTheory/Derivation/DifferentialRing.lean

```diff
@@ -0,0 +1,83 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.RingTheory.Derivation.Basic
+
+/-!
+# Differential Rings and Algebras
+
+This file defines differential rings, which are rings with a bundled `Derivation ℤ R R`, and
+differential algebras, which are algebras of differential rings where the algebra map commutes
+with the derivative.
+
+This file also defines the x′ notation for the derivative of x in a differential ring.
+-/
+
+/--
+A `CommRing` with a bundled `Derivation` to itself.
+-/
+class CommDifferentialRing (R : Type*) extends CommRing R where
```

Comment:
Are there any non-commutative `CommDifferentialRing`s?

Reply:
I'm not aware of any, the existing definition of Derivation requires commutativity, and Wikipedia defines it with commutativity, but I don't know differential algebra well enough to tell for sure.

Reply:
Would you ever want to consider multiple differential-ring structures on the same ring (maybe multiple partial derivatives)? If so, I would suggest `class DifferentialRing (R : Type*) [CommRing R] where` instead.

Reply:
You could have multiple `(d1 d2 : DifferentialRing R)` lying around, like how mathlib deals with multiple topologies on a type. Also you wouldn't be able to declare differential fields with this bundled approach.

Reply:
I would (and do, in #14860), they just extend both `CommDifferentialRing` and `Field`

Reply:
I tried to change it, what do you think? It's nicer in some ways, but it also requires writing more typeclass assumptions